src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -3242,7 +3247,8 @@ def stop(self):
         # During LRMS termination, we call any existing shutdown hooks on the
         # launch methods.  We only call LM shutdown hooks *once*
         launch_methods = set() # set keeps entries unique
-        launch_methods.add(self._cfg['mpi_launch_method'])
+        if 'mpi_launch_method' in self._cfg:
+            launch_methods.add(self._cfg['mpi_launch_method'])
         launch_methods.add(self._cfg['task_launch_method'])
```

Comment:
So far I've only made MPI tasks optional. If you want to do the same for regular tasks, that needs changes in more places.